channel/git.go

```diff
@@ -29,28 +29,27 @@ type Git struct {
 	mutex             *sync.Mutex
 }
 
-type GitVersions struct {
-	branches map[string]ConfigVersion
+type gitVersion struct {
+	git *Git
```

Comment:
nit: this circular dependency on `Git` which is also the thing returning `gitVersion` is a little bit weird. But since it's all internal I'm ok with it :)

Reply:
It's easier than copying most of the parts from the owning `Git`. I guess it could get an interface or something like a `func localClone(…)`, but I don't know if it'd improve the readability.